### PR TITLE
Fix access token login and related issues [Fixes #105057182]

### PR DESCRIPTION
**Pull request by markcowl**
- fixes access token login
- fixes serialization issues with profile
- fixes issues with data store

### CODE REVIEW

**Comment by markcowl** [comment]
on demand run:  http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/230/

File: src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs [hunk 1]
```diff
@@ -280,6 +283,12 @@ private AzureEnvironment MergeEnvironmentProperties(AzureEnvironment environment
             SecureString password,
             ShowDialog promptBehavior)
         {
+            if (account.Type == AzureAccount.AccountType.AccessToken)
+            {
+                tenantId = tenantId ?? "Common";
```
Comment (ogail) [inline src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs]: nit: can be moved to `SimpleAccessToken` and have an 2nd constructor that takes account and defaults tenant to `"Common"`

File: src/ResourceManager/Profile/Commands.Profile/Models/SimpleAccessToken.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            this.TenantId = tenantId;
+        }
+        public string AccessToken { get; private set; }
+
+        public void AuthorizeRequest(System.Action<string, string> authTokenSetter)
+        {
+            authTokenSetter(_tokenType, AccessToken);
+        }
+
+        public LoginType LoginType
```
Comment (ogail) [inline src/ResourceManager/Profile/Commands.Profile/Models/SimpleAccessToken.cs]: Nit: formatting
```csharp
public LoginType LoginType { get { return LoginType.OrgId; } }
```

File: src/ResourceManager/Profile/Commands.Profile/Models/SimpleAccessToken.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        {
+            get { return LoginType.OrgId; }
+        }
+
+        public string TenantId
```
Comment (ogail) [inline src/ResourceManager/Profile/Commands.Profile/Models/SimpleAccessToken.cs]: Nit: formatting
```csharp
public string TenantId { get; private set; }
```

File: src/ResourceManager/Profile/Commands.Profile/Models/SimpleAccessToken.cs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        {
+            get; 
+            private set;
+        }
+
+        public string UserId { get; private set; }
```
Comment (ogail) [inline src/ResourceManager/Profile/Commands.Profile/Models/SimpleAccessToken.cs]: Nit: add comments to the properties

**Comment by hovsepm** [comment]
on-demand 2: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/233/

**Pull request merged**